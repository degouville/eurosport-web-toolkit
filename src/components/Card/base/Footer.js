import PropTypes from 'prop-types';
import styled from 'react-emotion';
import { colors } from '../../..';

const Footer = styled('div')`
  display: flex;
  align-items: center;
  margin-top: 15px;
  border-top: 1px solid ${colors.whiteLilac};
  color: ${colors.santasGray};
  font-size: 12px;
`;

Footer.displayName = 'Footer';

Footer.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Footer;
