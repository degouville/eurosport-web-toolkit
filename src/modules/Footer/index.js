import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import * as breakpoints from '../../breakpoints';
import * as colors from '../../colors';
import Button from '../../elements/Button';

const StyledWrapper = styled.footer`
  ${breakpoints.medium(css`
    border-top: 1px solid rgba(255, 255, 255, 0.15);
    padding-top: 20px;
    display: flex;
    justify-content: space-between;
    align-items: flex-start;
  `)};
`;

const StyledButton = styled(Button)`
  font-size: 11px;
  line-height: 11px;
  ${breakpoints.medium(css`
    display: inline-block;
    padding: 11px 20px;
  `)};
`;

const StyledColumn = styled.div`
  font-size: 11px;
  text-align: center;
  margin-bottom: 20px;
  ${breakpoints.medium(css`
    text-align: right;
  `)};
`;

const StyledItems = styled.ul`
  margin: 30px 0 16px;
  padding-left: 0;
  ${breakpoints.medium(css`
    margin-top: 0;
  `)};
`;

const StyledItem = styled.li`
  list-style-type: none;
  border-top: 1px solid rgba(255, 255, 255, 0.15);

  &:last-of-type {
    border-bottom: 1px solid rgba(255, 255, 255, 0.15);
  }

  ${breakpoints.medium(css`
    display: inline-block;
    margin-left: 20px;
    border: none;

    &:last-of-type {
      border: none;
    }
  `)};
`;

const StyledLink = styled.a`
  padding: 15px 0;
  display: block;
  color: ${colors.athensGray};
  text-decoration: none;
  ${breakpoints.medium(css`
    display: inline-block;
    margin-left: 20px;
    padding: 0;
  `)};
  :hover {
    color: ${colors.white};
  }
`;

const StyledCopyright = styled.div`
  color: ${colors.manatee};
`;

const Footer = ({ faq, items, copyright, ...props }) => (
  <StyledWrapper {...props}>
    <StyledButton {...faq.linkProps} type="secondary" target="_blank">
      {faq.label}
    </StyledButton>
    <StyledColumn>
      <StyledItems>
        {items.map(
          item =>
            item.linkProps &&
            item.linkProps.href &&
            item.label && (
              <StyledItem key={item.id}>
                <StyledLink {...item.linkProps}>{item.label}</StyledLink>
              </StyledItem>
            )
        )}
      </StyledItems>
      {copyright && <StyledCopyright>{copyright}</StyledCopyright>}
    </StyledColumn>
  </StyledWrapper>
);

Footer.defaultProps = {
  items: [],
  copyright: null,
};

Footer.propTypes = {
  copyright: PropTypes.string,
  faq: PropTypes.shape({
    label: PropTypes.string,
    linkProps: PropTypes.shape({
      href: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      label: PropTypes.string,
      linkProps: PropTypes.shape({
        href: PropTypes.string.isRequired,
      }).isRequired,
    })
  ),
};

Footer.displayName = 'Footer';

export default Footer;
