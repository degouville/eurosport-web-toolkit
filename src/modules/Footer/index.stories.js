import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, object } from '@storybook/addon-knobs';
import { Footer } from '../..';

const footerStories = storiesOf('Modules|Footer', module);
const faq = {
  label: 'Help & Faq',
  linkProps: {
    href: 'https://google.fr',
  },
};
const items = [
  {
    id: 1,
    label: 'Terms & Conditions',
    linkProps: {
      href: 'https://google.com',
    },
  },
  {
    id: 2,
    label: 'Privacy Policy',
    linkProps: {
      href: 'https://google.com',
    },
  },
  {
    id: 3,
    label: 'Cookie Policy',
    linkProps: {
      href: 'https://google.com',
    },
  },
];

footerStories.add('base', () => (
  <Footer
    faq={object('faq', faq)}
    items={object('items', items)}
    copyright={text('copyright', '© Eurosport, a Discovery Company 2018 – All rights reserved')}
  />
));
