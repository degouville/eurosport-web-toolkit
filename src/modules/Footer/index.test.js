import React from 'react';
import { shallow } from 'enzyme';
import Footer from '.';

const faq = {
  label: 'Help & Faq',
  linkProps: {
    href: 'https://google.fr',
  },
};
const items = [
  {
    id: 1,
    label: 'Terms & Conditions',
    linkProps: {
      href: 'https://google.com',
    },
  },
  {
    id: 2,
    label: 'Privacy Policy',
    linkProps: {
      href: 'https://google.com',
    },
  },
  {
    id: 3,
    label: 'Cookie Policy',
    linkProps: {
      href: 'https://google.com',
    },
  },
];

describe('Footer', () => {
  it('renders when everything passed in', () => {
    expect(shallow(<Footer faq={faq} items={items} copyright="copyright" />)).toMatchSnapshot();
  });

  it('renders without copyright', () => {
    expect(shallow(<Footer faq={faq} items={items} />)).toMatchSnapshot();
  });

  it('renders with no items', () => {
    expect(shallow(<Footer faq={faq} copyright="copyright" />)).toMatchSnapshot();
  });
});
