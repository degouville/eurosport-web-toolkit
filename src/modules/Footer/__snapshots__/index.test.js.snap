// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders when everything passed in 1`] = `
<Styled(div)>
  <Styled(Link)
    href="https://google.fr"
    target="_blank"
    type="secondary"
  >
    Help & Faq
  </Styled(Link)>
  <Styled(div)>
    <Styled(ul)>
      <Styled(li)
        key="1"
      >
        <Styled(a)
          href="https://google.com"
        >
          Terms & Conditions
        </Styled(a)>
      </Styled(li)>
      <Styled(li)
        key="2"
      >
        <Styled(a)
          href="https://google.com"
        >
          Privacy Policy
        </Styled(a)>
      </Styled(li)>
      <Styled(li)
        key="3"
      >
        <Styled(a)
          href="https://google.com"
        >
          Cookie Policy
        </Styled(a)>
      </Styled(li)>
    </Styled(ul)>
    <Styled(div)>
      copyright
    </Styled(div)>
  </Styled(div)>
</Styled(div)>
`;

exports[`Footer renders with no items 1`] = `
<Styled(div)>
  <Styled(Link)
    href="https://google.fr"
    target="_blank"
    type="secondary"
  >
    Help & Faq
  </Styled(Link)>
  <Styled(div)>
    <Styled(ul) />
    <Styled(div)>
      copyright
    </Styled(div)>
  </Styled(div)>
</Styled(div)>
`;

exports[`Footer renders without copyright 1`] = `
<Styled(div)>
  <Styled(Link)
    href="https://google.fr"
    target="_blank"
    type="secondary"
  >
    Help & Faq
  </Styled(Link)>
  <Styled(div)>
    <Styled(ul)>
      <Styled(li)
        key="1"
      >
        <Styled(a)
          href="https://google.com"
        >
          Terms & Conditions
        </Styled(a)>
      </Styled(li)>
      <Styled(li)
        key="2"
      >
        <Styled(a)
          href="https://google.com"
        >
          Privacy Policy
        </Styled(a)>
      </Styled(li)>
      <Styled(li)
        key="3"
      >
        <Styled(a)
          href="https://google.com"
        >
          Cookie Policy
        </Styled(a)>
      </Styled(li)>
    </Styled(ul)>
  </Styled(div)>
</Styled(div)>
`;
